pkg/volume/host_path/host_path_test.go

```diff
@@ -190,9 +190,11 @@ func TestPlugin(t *testing.T) {
 	if err != nil {
 		t.Errorf("Can't find the plugin by name")
 	}
+
+	typeVol := v1.HostPathDirectoryOrCreate
 	spec := &v1.Volume{
 		Name:         "vol1",
-		VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/vol1"}},
+		VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/vol1", Type: &typeVol}},
```

Comment:
Now that you have the interface, you can mock it out and exhaustively test that every case works, which would have caughtthe bugs I flagged above :)